migrating_gazebo_classic_ros2_packages.md

```diff
@@ -0,0 +1,593 @@
+`turtlebot3_simulations` covered in this tutorial, can be found in
+[this fork](https://github.com/azeey/turtlebot3_simulations/tree/new_gazebo)
+
+We'll start by following the
+[PC Setup](https://emanual.robotis.com/docs/en/platform/turtlebot3/quick-start/)
+guide to install the necessary prerequisites for simulating Turtlebot3. This
+will install additional packages, such as Nav2 and Cartopgrapher, which we will
+be using later on in this tutorial, so make sure to not skip this step.
+
+The next step is to clone the `turtlebot3_simulation` package. We'll use the
+`humble-devel` branch, which at the time of writing had a SHA of `d16cdbe`
+
+```bash
+source /opt/ros/humble/setup.bash
+mkdir -p ~/turtlebot3_ws/src
+cd ~/turtlebot3_ws/src
+git clone -b d16cdbe https://github.com/ROBOTIS-GIT/turtlebot3_simulations/
+```
+
+Install dependencies using `rosdep`
+
+```bash
+sudo rosdep init # only needed if  using rosdep
+rosdep install --from-paths . -i -y
+```
+
+Finally, build the project and check that the Gazebo classic simulation works.
+(See the
+[Gazebo Simulation guide](https://emanual.robotis.com/docs/en/platform/turtlebot3/simulation/#gazebo-simulation))
+
+```bash
+cd ~/turtlebot3_ws
+colcon build --symlink-install
+source ~/turtlebot3_ws/install/setup.bash
+
+export TURTLEBOT3_MODEL=waffle
+ros2 launch turtlebot3_gazebo empty_world.launch.py
+```
+
+Once, we're sure that the Gazebo classic simulation is running properly, we
+create a new branch in which we'll make the changes to migrate to the new
+Gazebo.
+
+```bash
+git checkout -b new_gazebo
+```
+
+The changes we need to make are:
+
+1. Modify `package.xml` and `CMakeLists.txt` files replacing `gazebo`,
+   `gazebo_ros_pkgs`, etc with packages from `ros_gz`.
+1. Edit launch files that start Gazebo (e.g. `empty_world.launch.py`)
+1. Update the world SDFormat file.
+1. Edit launch files that spawn models.
+1. Edit model SDFormat files.
+1. Bridge ROS topics.
+
+## Update package dependencies
+
+The package depends on `gazebo_ros_pkgs`, which is the package that provides
+launch files, plugins, and other utilities for using Gazebo classic with ROS 2.
+The equivalent for the new Gazebo is `ros_gz`, but `ros_gz` is actually a
+meta-package that contains a few packages. It's okay to replace
+`gazebo_ros_pkgs` with `ros_gz` here, but using just the subset of packages
+needed for your project will reduce the number of dependencies. For the
+`turtlebot3_simulation` package, we will only need `ros_gz_bridge` and
+`ros_gz_sim` for now. `ros_gz_bridge` provides a topic bridge between Gazebo and
+ROS while `ros_gz_sim` provides launch files and other utilities that help with
+starting Gazebo and spawning models.
+
+After making the change, lines 17-21 of `package.xml` will look like this:
+
+```xml
+...
+  <buildtool_depend>ament_cmake</buildtool_depend>
+  <depend>ros_gz_bridge</depend>
+  <depend>ros_gz_sim</depend>
+  <depend>geometry_msgs</depend>
+  <depend>nav_msgs</depend>
```

Comment:
alphabetize

Reply:
Done in https://github.com/gazebosim/docs/pull/425/commits/90742e17a59476027637e42bde91d1ef25755b38